primitives/traits/src/lib.rs

```diff
@@ -615,12 +615,45 @@ pub trait ExternalIndexProvider {
     fn get_external_index() -> u64;
 }
 
-// A trait to verify if a node has been inactive during the last minimum activity
+// A trait to check invulnerables
+pub trait InvulnerablesHelper<AccountId> {
+    /// Checks if the given `AccountId` is invulnerable.
+    fn is_invulnerable(account_id: &AccountId) -> bool;
+}
+
+// A trait to verify the inactivity status of nodes
+// and handle the offline status of nodes
 pub trait NodeActivityTrackingHelper<AccountId> {
+    /// Check if a node is inactive.
     fn is_node_inactive(node: &AccountId) -> bool;
+    /// Check if a node is offline.
+    fn is_node_offline(node: &AccountId) -> bool;
+    #[cfg(feature = "runtime-benchmarks")]
+    /// Marks online node as online
+    fn make_node_online(node: &AccountId);
+    /// Marks node as inactive for the current activity window so it could be notified as inactive
+    #[cfg(feature = "runtime-benchmarks")]
+    fn make_node_inactive(node: &AccountId);
 }
 
 // A trait to help verify if a ParaId is a chain or parathread
 pub trait ParathreadHelper {
     fn get_parathreads_for_session() -> BTreeSet<ParaId>;
 }
+
+// A trait to help remove offline collators from the pending assignment
+pub trait PendingCollatorAssignmentHelper<AccountId> {
+    /// Remove a collator from the current pending collator assignment.
+    fn remove_offline_collator_from_pending_assignment(collator: &AccountId);
+}
+
+// A trait to update the collators rewards when a collator's online status changes.
+pub trait NotifyCollatorOnlineStatusChange<AccountId> {
+    /// Check if the collator is in SortedEligibleCandidates list.
+    fn is_collator_in_sorted_eligible_candidates(collator: &AccountId) -> bool;
```

Comment:
Make the function names a bit more agnostic? Something like `is_candidate_selected` and `on_online_status_change` ? In case other implementations needs it, maybe include the online status as a bool here ? (even if we don't need it for pooled staking)